compiler-rt/lib/scudo/standalone/secondary.h

```diff
@@ -807,7 +821,13 @@ void MapAllocator<Config>::deallocate(const Options &Options, void *Ptr)
     Stats.sub(StatAllocated, CommitSize);
     Stats.sub(StatMapped, H->MemMap.getCapacity());
   }
-  Cache.store(Options, H);
+
+  if (!Cache.canCache(H->CommitSize)) {
```

Comment:
Small nit, it is usually slightly easier to read if you have the positive case first and the negative case second. Thus do the sotre in the if clause and the unmap in the else.